src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//==== ONNXToTosaLegalizeUtils.cpp - ONNX dialects to TOSA lowering Utils-===//
+//
+// Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file contains common utils shared by the functions performing the
+// lowering to the TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Tosa/Utils/QuantUtils.h"
+#include "mlir/Dialect/Tosa/Utils/ShapeUtils.h"   // from @llvm-project
+#include "mlir/IR/BuiltinAttributes.h"            // from @llvm-project
+#include "mlir/IR/BuiltinTypes.h"                 // from @llvm-project
+#include "mlir/IR/PatternMatch.h"                 // from @llvm-project
+#include "mlir/Interfaces/InferTypeOpInterface.h" // from @llvm-project
+#include "mlir/Support/LLVM.h"
+
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.hpp" // from @llvm-project
+#include "llvm/ADT/STLExtras.h"
+#include "llvm/ADT/SmallVector.h"
+#include "llvm/IR/DerivedTypes.h"
+#include <cstdint>
+#include <src/Dialect/Mlir/IndexExpr.hpp>
+
+using namespace mlir;
+namespace onnx_mlir {
+namespace tosa {
+
+Value createTosaTransposedTensor(PatternRewriter &rewriter, Operation *op,
+    Value &value, llvm::ArrayRef<int64_t> perm) {
+  // Create Permutation Const
+  Value permList = getConstTensor(
+      rewriter, op, perm, {value.getType().cast<RankedTensorType>().getRank()})
+                       .value();
+  auto valueType = value.getType().cast<ShapedType>();
+  // get new value type
+  Type newValueType = RankedTensorType::get(
+      llvm::SmallVector<int64_t, 4>(valueType.getShape().size(), -1),
+      valueType.getElementType());
+  // create transpose for value
+  Value newValue = CreateOpAndInfer<mlir::tosa::TransposeOp>(
```

Comment:
Great pattern, we were looking into doing something along these line, will look at the code to see if we can reuse the code or pattern

Reply:
Thank you!